File: Python/BalancedStringSplit.py
def balancedStringSplit(s):
    count = 0
    countl = 0
    countr = 0
    flag = True
    for i in range(0, len(s), 1):
        if (i > 0) and (s[i] != s[i - 1]):
            if flag:
                flag = False
            else:
                flag = True
        if flag:
            countl += 1
        if not flag:
            countr += 1
        if countl == countr:
            count += 1
            countl = 0
            countr = 0
    return count


s = "RLRRRLLRLL"
out = balancedStringSplit(s)
print(out)

File: C++/BalancedStringSplit.cpp
//#include <iostream>
//#include <vector>
//#include <string>
//using namespace std;
//
//int balancedStringSplit(string s) {
//	int count = 0;
//	int countl = 0;
//	int countr = 0;
//	bool flag = true;
//	for (int i = 0; i < s.size(); i++) {
//		if ((i > 0) && (s[i] != s[i - 1])) {
//			if (flag == true) flag = false;
//			else flag = true;
//		}
//		if (flag) {
//			countl++;
//		}
//		if (!flag) {
//			countr++;
//		}
//		if (countl == countr) {
//			countl = 0;
//			countr = 0;
//			count++;
//		}
//	}
//	return count;
//}
//
//int main() {
//	string s = "RLRRLLRLRL";
//	int out;
//	out = balancedStringSplit(s);
//	cout << out << endl;
//	return 0;
//}